snuba/request/schema.py

```diff
@@ -63,7 +64,7 @@ def validate(self, value) -> Request:
             extensions[extension_name] = {key: value.pop(key) for key in extension_schema['properties'].keys() if key in value}
 
         return Request(
-            Query(query_body),
+            Query(query_body, data_source),
```

Comment:
I am not yet sure about how this would look like in the longer term. As of now we do not pick the dataset from the body, so it is fair that it is passed to validate.
I think this will be revisited when introducing the Column abstraction. But at this point I don't think I have enough clear information yet to refactor this validation process.